packages/@react-spectrum/table/test/TestTableUtils.test.js

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {act, render, screen} from '@react-spectrum/test-utils-internal';
+import {Cell, Column, Row, TableBody, TableHeader, TableView} from '../';
+import {Provider} from '@react-spectrum/provider';
+import React, {useState} from 'react';
+import {theme} from '@react-spectrum/theme-default';
+import {User} from '@react-aria/test-utils';
+
+let manyItems = [];
+for (let i = 1; i <= 100; i++) {
+  manyItems.push({id: i, foo: 'Foo ' + i, bar: 'Bar ' + i, baz: 'Baz ' + i});
+}
+
+let columns = [
+  {name: 'Foo', key: 'foo'},
+  {name: 'Bar', key: 'bar'},
+  {name: 'Baz', key: 'baz'}
+];
+
+describe('Table ', function () {
+  let onSelectionChange = jest.fn();
+  let onSortChange = jest.fn();
+  let testUtilRealTimer = new User();
+
+  let TableExample = (props) => {
+    let [sort, setSort] = useState({});
+    let setSortDescriptor = (sort) => {
+      setSort(sort);
+      onSortChange(sort);
+    };
+
+    return (
+      <Provider theme={theme}>
+        <TableView aria-label="Table" selectionMode="multiple" data-testid="test" sortDescriptor={sort} onSortChange={setSortDescriptor} onSelectionChange={onSelectionChange} {...props}>
+          <TableHeader columns={columns}>
+            {column => <Column allowsSorting allowsResizing={props.allowsResizing}>{column.name}</Column>}
+          </TableHeader>
+          <TableBody items={manyItems}>
+            {item =>
+              (<Row key={item.foo}>
+                {key => <Cell>{item[key]}</Cell>}
+              </Row>)
+            }
+          </TableBody>
+        </TableView>
+      </Provider>
+    );
+  };
+
+  describe.each`
+    interactionType
+    ${'mouse'}
+    ${'keyboard'}
+    ${'touch'}
+  `('with real timers, interactionType: $interactionType ', ({interactionType}) => {
```

Comment:
This is sort of what I'm thinking about above with select keyboard open triggers, but this is interaction types. This seems like a good way to think about interaction types for all components. Probably a future enhancement refactor.

Reply:
yeah, these are pretty cool patterns possible with @LFDanLu 's utilities. I've expanded on it with a more generalized solution here https://github.com/adobe/react-spectrum/pull/7011/files#diff-d1f01b7d4c84a647fb746a881fe57008c94b007ecf5c98f8308c210901b091e0R20